### PR TITLE
Close #102, save testing files in da 'sources' folder

### LINKED ISSUE #102: Save tests to `Sources` folder
Developers can update their files in docassemble at the same time as they update their interview, possibly avoiding unnecessarily failing tests.

~If the admin does this, developers can put their testing files in the Sources folder and edit them there instead of in GitHub.~

### PR DESCRIPTION
**Pull request by plocket**
Closes #102. Lets developers update tests at the same time as updating their interviews. See passing action and check folder locations: https://github.com/plocket/docassemble-ALAutomatedTestingTests/actions/runs/741358145

Comes after https://github.com/plocket/docassemble-cucumber/pull/213

Also tweaks some other wording and presentation.

### CODE REVIEW

File: docassemble/ALAutomatedTestingTests/data/questions/al_set_up_testing.yml [hunk 1]
```diff
@@ -48,6 +48,10 @@ code: |
   #instructions
   next_steps
 ---
+code: |
+  installer.set_da_info()
+  set_da_info = True
+---
```
Comment (plocket) [inline docassemble/ALAutomatedTestingTests/data/questions/al_set_up_testing.yml:54]: Unrelated, sorry. Moved code block to be with other code blocks.

**Comment by plocket** [comment]
I tested this with a few setup branches that I've turned into drafts. I think that for some reason something got messed up with the secrets and that stopped setup from happening correctly. Not sure what was going on there. Editing the secrets manually seemed to fix it. I ran setup one more time and it stayed fixed. That will have to be explored, but for now things seem to be working correctly.

File: docassemble/ALAutomatedTestingTests/data/questions/al_testing_files_to_push.yml [hunk 2]
```diff
@@ -70,7 +70,7 @@ content: |
     "scripts": {
       "test": "npm run langs_setup && npm run cucumber -- $@",
       "cucumber_base": "./node_modules/.bin/cucumber-js --require ./node_modules/docassemble-cucumber/lib/index.js",
-      "cucumber": "run_cucumber(){ npm run cucumber_base -- \"$@\" tests/features/*.feature; }; run_cucumber",
+      "cucumber": "run_cucumber(){ npm run cucumber_base -- \"$@\" tests/features/*.feature docassemble/*/data/sources/*.feature; }; run_cucumber",
```
Comment (niharikasingh) [inline docassemble/ALAutomatedTestingTests/data/questions/al_testing_files_to_push.yml:73]: Hmm... do we still need `tests/features/*.feature` in here? You removed it in https://github.com/plocket/docassemble-cucumber/pull/213/commits/f70dd47a9cfe6c2864f4e7ced48c6ef1f0992869 but it's back here now.

File: docassemble/ALAutomatedTestingTests/al_set_up_testing.py [hunk 3]
```diff
@@ -195,11 +196,12 @@ def push_files( self ):
     """Send files to folders in new branch in github.
     https://pygithub.readthedocs.io/en/latest/examples/Repository.html#create-a-new-file-in-the-repository'''
     # https://pygithub.readthedocs.io/en/latest/github_objects/Repository.html#github.Repository.Repository.create_file"""
+    test_path = 'docassemble/' + self.package_name + '/data/sources/example_test.feature'
+    self.send_file( test_path, 'Add tests/features/example_test.feature', self.example_test_str )  # 2
```
Comment (niharikasingh) [inline docassemble/ALAutomatedTestingTests/al_set_up_testing.py:200]: Should `example_test.feature` still exist in `tests/features` or should that be moved to `sources` too? (Not urgent, but at some point.)

**Review by niharikasingh — APPROVED** [review]

File: docassemble/ALAutomatedTestingTests/data/questions/al_testing_files_to_push.yml (hunk 2 above)
Comment (plocket) [reply docassemble/ALAutomatedTestingTests/data/questions/al_testing_files_to_push.yml:73]: Thank you so much for catching this! No idea how this is here. Maybe I didn't push something.

File: docassemble/ALAutomatedTestingTests/al_set_up_testing.py (hunk 3 above)
Comment (plocket) [reply docassemble/ALAutomatedTestingTests/al_set_up_testing.py:200]: Ah, the commit message needs to be updated. Thank you!

**Comment by plocket** [comment]
Apparently I didn't actually push the updates. They should be here now, but a sanity check would be nice.

**Comment by plocket** [comment]
I'll go ahead and merge it and we'll deal with the consequences!

**Pull request merged**